### PR TITLE
Support for Ecto 3.2

### PR DESCRIPTION
**Pull request by youalreadydid**
Hi, this PR aims to support Ecto 3.2.

Besides adding support for `mix format`, I've added `embed_as/1` and `equal?/2` to Ecto.Type implementations (Action, Patch, Schema) and `prepare_query/3` to Repo Queryable.

All tests are passing with Elixir 1.9 and Erlang 22, and I've tested it too with my company's private repos that already use ex_audit 0.6.

### CODE REVIEW

File: lib/repo/schema_type.ex [hunk 1]
```diff
@@ -39,4 +40,8 @@ defmodule ExAudit.Type.Schema do
   defp schemas do
     Application.get_env(:ex_audit, :tracked_schemas, [])
   end
+
+  def embed_as(_), do: :self
+  def equal?(term1, term2), do: term1 == term2
+  defoverridable embed_as: 1, equal?: 2
```
Comment (narrowtux) [inline lib/repo/schema_type.ex]: Isn't this unnecessary when not defining a behaviour?

File: lib/repo/action_type.ex [hunk 2]
```diff
@@ -26,4 +26,8 @@ defmodule ExAudit.Type.Action do
   def cast(_), do: :error
 
   def type, do: :string
-end
\ No newline at end of file
+
+  def embed_as(_), do: :self
+  def equal?(term1, term2), do: term1 == term2
+  defoverridable embed_as: 1, equal?: 2
```
Comment (narrowtux) [inline lib/repo/action_type.ex]: see other comment about `defoverridable`

File: lib/repo/patch_type.ex [hunk 3]
```diff
@@ -5,4 +5,8 @@ defmodule ExAudit.Type.Patch do
   def dump(patch), do: {:ok, :erlang.term_to_binary(patch)}
   def load(binary), do: {:ok, :erlang.binary_to_term(binary)}
   def type, do: :binary
-end
\ No newline at end of file
+
+  def embed_as(_), do: :self
+  def equal?(term1, term2), do: term1 == term2
+  defoverridable embed_as: 1, equal?: 2
```
Comment (narrowtux) [inline lib/repo/patch_type.ex]: see other comment about `defoverridable`

File: lib/repo/schema_type.ex (hunk 1 above)
Comment (youalreadydid) [reply lib/repo/schema_type.ex]: Nice catch :)

In [Ecto.Type code](https://github.com/elixir-ecto/ecto/blob/v3.2.5/lib/ecto/type.ex#L83) there is an `__using__/1` method that already call `@behaviour` and define this two functions. They keep the defoverridable, so it seems that it really works if someone wants to redefine the methods in a "extended" type. The [docs](https://hexdocs.pm/ecto/Ecto.Type.html) also suggest `use Ecto.Type`, so I've changed to comply with it.

**Comment by narrowtux** [comment]
Nice work

**Comment by narrowtux** [comment]
I just published v 0.7

**Pull request merged**